exchanges/lbank/lbank.go

```diff
@@ -0,0 +1,618 @@
+package lbank
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wshandler"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+// Lbank is the overarching type across this package
+type Lbank struct {
+	exchange.Base
+	privateKey    *rsa.PrivateKey
+	WebsocketConn *wshandler.WebsocketConnection
+}
+
+const (
+	lbankAPIURL          = "https://api.lbkex.com"
+	lbankAPIVersion      = "1"
+	lbankAuthRateLimit   = 0
+	lbankUnAuthRateLimit = 0
+	lbankFeeNotFound     = 0.0
+
+	// Public endpoints
+	lbankTicker         = "ticker.do"
+	lbankCurrencyPairs  = "currencyPairs.do"
+	lbankMarketDepths   = "depth.do"
+	lbankTrades         = "trades.do"
+	lbankKlines         = "kline.do"
+	lbankPairInfo       = "accuracy.do"
+	lbankUSD2CNYRate    = "usdToCny.do"
+	lbankWithdrawConfig = "withdrawConfigs.do"
+
+	// Authenticated endpoints
+	lbankUserInfo                = "user_info.do"
+	lbankPlaceOrder              = "create_order.do"
+	lbankCancelOrder             = "cancel_order.do"
+	lbankQueryOrder              = "orders_info.do"
+	lbankQueryHistoryOrder       = "orders_info_history.do"
+	lbankOrderTransactionDetails = "order_transaction_detail.do"
+	lbankPastTransactions        = "transaction_history.do"
+	lbankOpeningOrders           = "orders_info_no_deal.do"
+	lbankWithdrawalRecords       = "withdraws.do"
+	lbankWithdraw                = "withdraw.do"
+	lbankRevokeWithdraw          = "withdrawCancel.do"
+)
+
+// SetDefaults sets the basic defaults for Lbank
+func (l *Lbank) SetDefaults() {
+	l.Name = "Lbank"
+	l.RESTPollingDelay = 10
+	l.RequestCurrencyPairFormat.Delimiter = "_"
+	l.ConfigCurrencyPairFormat.Delimiter = "_"
+	l.AssetTypes = []string{ticker.Spot}
```

Comment:
Please add l.SupportsAutoPairUpdating = true as this exchange supports this feature and also set l.APIWithdrawPermissions. It looks like crypto withdrawals are supported by the API but fiat related ones arent so this will need to be adjusted accordingly.